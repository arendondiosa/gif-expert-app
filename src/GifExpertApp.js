import React, { useState } from "react";
import PropTypes from "prop-types";
import { Container, Jumbotron, Navbar } from "react-bootstrap";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <Container>
      <Navbar fixed="top" bg="dark" variant="dark">
        <Navbar.Brand href="#home">GifExpertApp</Navbar.Brand>
        <Navbar.Toggle />
      </Navbar>
      <div className="content">
        <Jumbotron>
          <h1>GifExpertApp</h1>
          <p>
            Search all the GIFs and Stickers. <code>Powered by GIPHY</code>
          </p>
          <AddCategory setCategories={setCategories} />
        </Jumbotron>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </div>
    </Container>
  );
};

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array,
};

GifExpertApp.defaultProps = {
  defaultCategories: [],
};
