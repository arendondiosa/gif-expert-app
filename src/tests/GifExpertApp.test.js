import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Tests <GifExpertApp />", () => {
  test("Render the component", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should display a list of categories", () => {
    const categories = ["Simpsons", "Batman", "Superman"];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
