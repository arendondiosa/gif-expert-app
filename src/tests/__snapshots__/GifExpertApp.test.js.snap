// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <GifExpertApp /> Render the component 1`] = `
<Container
  fluid={false}
>
  <Navbar
    bg="dark"
    collapseOnSelect={false}
    expand={true}
    fixed="top"
    variant="dark"
  >
    <NavbarBrand
      href="#home"
    >
      GifExpertApp
    </NavbarBrand>
    <NavbarToggle
      label="Toggle navigation"
    />
  </Navbar>
  <div
    className="content"
  >
    <Jumbotron
      fluid={false}
    >
      <h1>
        GifExpertApp
      </h1>
      <p>
        Search all the GIFs and Stickers. 
        <code>
          Powered by GIPHY
        </code>
      </p>
      <AddCategory
        setCategories={[Function]}
      />
    </Jumbotron>
  </div>
</Container>
`;

exports[`Tests <GifExpertApp /> Should display a list of categories 1`] = `
<Container
  fluid={false}
>
  <Navbar
    bg="dark"
    collapseOnSelect={false}
    expand={true}
    fixed="top"
    variant="dark"
  >
    <NavbarBrand
      href="#home"
    >
      GifExpertApp
    </NavbarBrand>
    <NavbarToggle
      label="Toggle navigation"
    />
  </Navbar>
  <div
    className="content"
  >
    <Jumbotron
      fluid={false}
    >
      <h1>
        GifExpertApp
      </h1>
      <p>
        Search all the GIFs and Stickers. 
        <code>
          Powered by GIPHY
        </code>
      </p>
      <AddCategory
        setCategories={[Function]}
      />
    </Jumbotron>
    <GifGrid
      category="Simpsons"
      key="Simpsons"
    />
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Superman"
      key="Superman"
    />
  </div>
</Container>
`;
